integration/dockertestutil/logs.go

```diff
@@ -40,6 +34,25 @@ func SaveLog(
 			Timestamps:   false,
 		},
 	)
+}
+
+func SaveLog(
+	pool *dockertest.Pool,
+	resource *dockertest.Resource,
+	basePath string,
+) (string, string, error) {
+	err := os.MkdirAll(basePath, os.ModePerm)
+	if err != nil {
+		return "", "", err
+	}
+
+	// Wouldn't it be simpler to
+	// open and wrap the destination files in a
+	// bufio.Writer, and pass those in docker.LogsOptions?
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
```

Comment:
Potentially, not tested or was aware of it, happy for that to be changed, maybe you can test in a followup pr?

Reply:
I haven't tested it either, I just thought maybe doing the equivalent of `> output.file` is simpler than doing an extra lap around the same thing. I don't expect any significant improvements or breakages from it either, other than the code becoming shorter. I don't mind doing an extra PR, but should it include only this spot, or all other file operations that may involve redundant `io.Writer`s?

Reply:
If you have time, all would of course be useful. Since this is in the integration tests, its not really that critical, but always nice to have it neater.

Reply:
I tried to look for other manually buffered disk writes with `git grep`, but as far as I can see, this was the only spot where it happened.

I can submit an MR with the changes, but... the code did not become much shorter. The tally comes to -5 lines, three of which were the comment I added. I could *still* see some value in it, because unlike `bytes.Buffer`, which (at least if I remember correctly) by default extends however many times it needs to fit things into memory, `bufio.Writer` has a defined buffer size and automatically flushes when it is about to cross it. So, in theory it *could* make it so that longer logs can be dumped without issue. I'm still waiting for the [integration tests to run](https://github.com/enoperm/headscale/pull/4) on github, but I do not expect problems with it.

(My machine does not seem to like the idea of running the entire test suite at once too much - best I could do was 34 to go before it timed out, and I think I might have raised the timeout before reaching that result).

Reply:
Aaaand it actually failed ina few tests - must have forgot about a flag.

Reply:
fair, this isnt really critical, if it isnt much simpler or neater, it can be dropped. I'm happy either way as it currently does work, good suggestion tho. You can submit it if it seem to work, and we can take if from there but you dont have to spend that much time on it.

Reply:
Removed the comment for now. I *may* take another look at it, but at this point I would not reallly mind leaving it as it is either.